{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After filtering the dataset with vcftools, we need (to pre-filter data in order to have a clean vcf or) to convert the .vcf file to .hdf5. \n",
    "note: Check that the ipyrad environment that you are using has a version of \n",
    "python between 3.5 to 3.9 (the version 3.10 doesn't work well in the moment)\n",
    "\n",
    "#conda create -n ipyrad_env python=3.9\n",
    "#conda activate ipyrad_env\n",
    "#conda install ipyrad -c conda-forge -c bioconda\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#the analysis tools are a subpackage of ipyrad\n",
    "#import ipyrad as ip\n",
    "#import ipyrad.analysis as ipa\n",
    "#import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Importing modules\n",
    "import ipyrad\n",
    "import ipyrad.analysis as ipa      ## ipyrad analysis toolkit\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Converting vcf to HDF5 using default ld_block_size: 20000\n",
      "Typical RADSeq data generated by ipyrad/stacks will ignore this value.\n",
      "You can use the ld_block_size parameter of the PCA() constructor to change\n",
      "this value.\n",
      "\n",
      "Indexing VCF to HDF5 database file\n",
      "VCF: 2189 SNPs; 181 scaffolds\n",
      "[                    ]   0% 0:00:00 | converting VCF to HDF5 \n",
      "This appears to be a denovo assembly, ld_block_size arg is being ignored.\n",
      "[####################] 100% 0:00:06 | converting VCF to HDF5 \n",
      "HDF5: 2189 SNPs; 181 linkage group\n",
      "SNP database written to ./analysis-vcf2hdf5/Alpsmac_linked_95th.recode.snps.hdf5\n",
      "Samples: 190\n",
      "Sites before filtering: 2189\n",
      "Filtered (indels): 0\n",
      "Filtered (bi-allel): 52\n",
      "Filtered (mincov): 0\n",
      "Filtered (minmap): 0\n",
      "Filtered (subsample invariant): 0\n",
      "Filtered (minor allele frequency): 0\n",
      "Filtered (combined): 52\n",
      "Sites after filtering: 2137\n",
      "Sites containing missing values: 2137 (100.00%)\n",
      "Missing values in SNP matrix: 75881 (18.69%)\n",
      "SNPs (total): 2137\n",
      "SNPs (unlinked): 181\n",
      "Imputation (null; sets to 0): 100.0%, 0.0%, 0.0%\n"
     ]
    }
   ],
   "source": [
    "## Converting .vcf file to .hdf5 file\n",
    "vcffile = \"/dss/dsshome1/0A/ra93rey/final_datasets/Alpsmac_linked_95th.recode.vcf\"\n",
    "pca = ipa.pca(vcffile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "vscode": {
     "languageId": "bat"
    }
   },
   "outputs": [],
   "source": [
    "\n",
    "##Converting clean vcf to hdf5\n",
    "\n",
    "# init a conversion tool. ld_block_size: size of the linkage blocks in base pairs\n",
    "converter = ipa.vcf_to_hdf5(\n",
    " name=\"Alps_wocgn_7add\",\n",
    " data=\"/dss/dsshome1/0A/ra93rey/ipyrad/cloud_PCA/alps_wocgn_7added.cleaned.vcf.gz\",\n",
    " ld_block_size=20000,\n",
    ")\n",
    "# run the converter\n",
    "converter.run()\n",
    "\n",
    "Indexing VCF to HDF5 database file\n",
    "VCF: 6094152 SNPs; 1 scaffolds\n",
    "[####################] 100% 0:02:22 | converting VCF to HDF5\n",
    "HDF5: 6094152 SNPs; 10845 linkage group\n",
    "SNP database written to ./analysis-vcf2hdf5/Macaque_LD20K.snps.hdf5\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "vscode": {
     "languageId": "powershell"
    }
   },
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (937971825.py, line 2)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  Input \u001b[0;32mIn [8]\u001b[0;36m\u001b[0m\n\u001b[0;31m    conda install scikit-learn -c bioconda\u001b[0m\n\u001b[0m          ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "#installation of the required software\n",
    "# conda install ipyrad -c bioconda\n",
    "# conda install scikit-learn -c bioconda\n",
    "# conda install toyplot -c eaton-lab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "#the analysis tools are a subpackage of ipyrad\n",
    "import ipyrad.analysis as ipa\n",
    "import pandas as pd\n",
    "import toyplot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "## making 2 dictionaries (species_dict and loc_dict) to use in PCA \n",
    "with open(\"/dss/dsshome1/0A/ra93rey/ipyrad/cloud_PCA/porous-alp-popdist-scheme-184indv.txt\") as file:\n",
    "    file.readline() #reads and discards the 1st line of the file (headers)\n",
    "    content = file.read() #reads the rest of the file's content and stores it in the variable content\n",
    "\n",
    "    species_dict = {} #empty dictionary named species_dict\n",
    "    loc_dict = {} #empty dictionary named loc_dict\n",
    "\n",
    "## 0 corresponds to id ,1 to locations, 7 to species (number of the columns)\n",
    "#The code is splitting the data into lines and then further splitting each line by tab (\"\\t\") to extract information. \n",
    "for line in content.splitlines():\n",
    "    line_list = line.split(\"\\t\")\n",
    "    if len(line_list) == 0 or len(line_list[0])  == 0 : #check if the line is empty\n",
    "        continue\n",
    "    else:\n",
    "        if line_list[7].strip() not in species_dict:\n",
    "            species_dict[line_list[7].strip()] = [line_list[0]]\n",
    "        else:\n",
    "            species_dict[line_list[7].strip()].append(line_list[0])\n",
    "        if line_list[1].strip() not in loc_dict:\n",
    "            loc_dict[line_list[1].strip()] = [line_list[0]]\n",
    "        else:\n",
    "            loc_dict[line_list[1].strip()].append(line_list[0])\n",
    "\n",
    "## now we have the dictionaries!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#import input data file .snps.hdf5 via file path\n",
    "data = \"/dss/dsshome1/0A/ra93rey/ipyrad/cloud_PCA/alps_wocgn_7added.snps.hdf5\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create an imap file (like a dictionary)\n",
    "#group individuals into populations\n",
    "#species_dict\n",
    "imap = species_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create a minmap dictionary that require that 20% of samples have data in each group (20% of coverage in each pop). This filters SNPs to only include those that have data for at least some proportion of samples in every group.\n",
    "minmap = {i: 0.2 for i in imap}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Kmeans clustering: iter=0, K=3, mincov=0.9, minmap={'global': 0.2}\n",
      "Samples: 184\n",
      "Sites before filtering: 7238146\n",
      "Filtered (indels): 143258\n",
      "Filtered (bi-allel): 222874\n",
      "Filtered (mincov): 7236776\n",
      "Filtered (minmap): 6981275\n",
      "Filtered (subsample invariant): 375268\n",
      "Filtered (minor allele frequency): 0\n",
      "Filtered (combined): 7238121\n",
      "Sites after filtering: 1180\n",
      "Sites containing missing values: 1180 (100.00%)\n",
      "Missing values in SNP matrix: 13714 (6.32%)\n",
      "SNPs (total): 1180\n",
      "SNPs (unlinked): 39\n",
      "Imputation: 'sampled'; (0, 1, 2) = 95.1%, 4.2%, 0.7%\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/dss/dsshome1/0A/ra93rey/.conda/envs/ipyrad/lib/python3.10/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{0: ['RP1230', 'RP1231', 'RP1232', 'RP1233', 'RP1234', 'RP1238', 'RP1240', 'RP1253', 'RP1254', 'RP1256', 'RP1257', 'RP1258', 'RP1259', 'RP1260', 'RP1261', 'RP1263', 'RP1264', 'RP1265', 'RP1266', 'RP1267', 'RP1268', 'RP1269', 'RP1270', 'RP1271', 'RP1272', 'RP1274', 'RP1275', 'RP1276', 'RP1277', 'RP1278', 'RP1279', 'RP1280', 'RP1281', 'RP1283', 'RP1284', 'RP1285', 'RP1287', 'RP1290', 'RP1291', 'RP1293', 'RP1294', 'RP1295', 'RP1297', 'RP1301', 'RP1302', 'RP1312', 'RP1379', 'RP1403', 'RP1404', 'RP1405', 'RP1406', 'RP1407', 'RP1417', 'RP1418', 'RP1419', 'RP1420', 'RP1421', 'RP1422', 'RP1423', 'RP1426', 'RP1427', 'RP1456', 'RP1458', 'RP1459', 'RP1461', 'RP1462', 'RP1479', 'RP1480', 'RP1481', 'RP1482', 'RP1484', 'RP1485', 'RP1487', 'RP1488', 'RP1489', 'RP1490', 'RP1491', 'RP1494', 'RP1495', 'RP1496', 'RP1497', 'RP1499', 'RP1508', 'RP1510', 'RP1511', 'RP1512', 'RP1530', 'RP1550', 'RP1551', 'RP1561', 'RP1562', 'RP1563', 'RP1564', 'RP1581', 'RP1582', 'RP1583', 'RP1584'], 1: ['RP1210', 'RP1211', 'RP1212', 'RP1213', 'RP1221', 'RP1239', 'RP1255', 'RP1282', 'RP1286', 'RP1288', 'RP1289', 'RP1292', 'RP1298', 'RP1304', 'RP1305', 'RP1313', 'RP1314', 'RP1315', 'RP1316', 'RP1317', 'RP1338', 'RP1339', 'RP1340', 'RP1341', 'RP1342', 'RP1343', 'RP1380', 'RP1381', 'RP1382', 'RP1383', 'RP1384', 'RP1385', 'RP1386', 'RP1388', 'RP1434', 'RP1436', 'RP1437', 'RP1438', 'RP1439', 'RP1441', 'RP1442', 'RP1460', 'RP1463', 'RP1464', 'RP1465', 'RP1466', 'RP1467', 'RP1468', 'RP1470', 'RP1483', 'RP1509', 'RP1513', 'RP1517', 'RP1520', 'RP1521', 'RP1531', 'RP1532', 'RP1533', 'RP1534', 'RP1549', 'RP1565', 'RP1575', 'RP1576', 'RP1577', 'RP1578', 'RP1579'], 2: ['I1', 'I2', 'I3', 'M1', 'RP1361', 'RP1362', 'RP1363', 'RP1364', 'RP1365', 'RP1366', 'RP1369', 'RP1552', 'RP1553', 'RP1554', 'RP1555', 'RP1556', 'RP1569', 'RP1570', 'RP1571', 'RP1572', 'RP1573']}\n",
      "\n",
      "Kmeans clustering: iter=1, K=3, mincov=0.7250000000000001, minmap={0: 0.2, 1: 0.2, 2: 0.2}\n",
      "Samples: 184\n",
      "Sites before filtering: 7238146\n",
      "Filtered (indels): 143258\n",
      "Filtered (bi-allel): 222874\n",
      "Filtered (mincov): 7231062\n",
      "Filtered (minmap): 7134123\n",
      "Filtered (subsample invariant): 375268\n",
      "Filtered (minor allele frequency): 0\n",
      "Filtered (combined): 7233223\n",
      "Sites after filtering: 6078\n",
      "Sites containing missing values: 6078 (100.00%)\n",
      "Missing values in SNP matrix: 188224 (16.83%)\n",
      "SNPs (total): 6078\n",
      "SNPs (unlinked): 193\n",
      "Imputation: 'sampled'; (0, 1, 2) = 95.2%, 4.1%, 0.8%\n",
      "{0: ['I1', 'I2', 'I3', 'M1', 'RP1361', 'RP1362', 'RP1363', 'RP1364', 'RP1365', 'RP1366', 'RP1369', 'RP1552', 'RP1553', 'RP1554', 'RP1555', 'RP1556', 'RP1569', 'RP1570', 'RP1571', 'RP1572', 'RP1573'], 1: ['RP1230', 'RP1231', 'RP1232', 'RP1233', 'RP1234', 'RP1238', 'RP1240', 'RP1253', 'RP1254', 'RP1255', 'RP1256', 'RP1257', 'RP1258', 'RP1259', 'RP1260', 'RP1261', 'RP1263', 'RP1264', 'RP1265', 'RP1266', 'RP1267', 'RP1268', 'RP1269', 'RP1270', 'RP1271', 'RP1272', 'RP1274', 'RP1275', 'RP1276', 'RP1277', 'RP1278', 'RP1279', 'RP1280', 'RP1281', 'RP1282', 'RP1283', 'RP1284', 'RP1285', 'RP1286', 'RP1287', 'RP1288', 'RP1289', 'RP1290', 'RP1291', 'RP1292', 'RP1293', 'RP1294', 'RP1295', 'RP1297', 'RP1298', 'RP1301', 'RP1302', 'RP1403', 'RP1404', 'RP1405', 'RP1406', 'RP1407', 'RP1417', 'RP1418', 'RP1419', 'RP1420', 'RP1421', 'RP1422', 'RP1423', 'RP1426', 'RP1427', 'RP1456', 'RP1458', 'RP1459', 'RP1461', 'RP1462', 'RP1479', 'RP1480', 'RP1481', 'RP1482', 'RP1483', 'RP1484', 'RP1485', 'RP1487', 'RP1488', 'RP1489', 'RP1490', 'RP1494', 'RP1495', 'RP1496', 'RP1497', 'RP1499', 'RP1508', 'RP1509', 'RP1510', 'RP1511', 'RP1512', 'RP1513', 'RP1530', 'RP1550', 'RP1551', 'RP1561', 'RP1562', 'RP1563', 'RP1564', 'RP1565', 'RP1581', 'RP1582', 'RP1583', 'RP1584'], 2: ['RP1210', 'RP1211', 'RP1212', 'RP1213', 'RP1221', 'RP1239', 'RP1304', 'RP1305', 'RP1312', 'RP1313', 'RP1314', 'RP1315', 'RP1316', 'RP1317', 'RP1338', 'RP1339', 'RP1340', 'RP1341', 'RP1342', 'RP1343', 'RP1379', 'RP1380', 'RP1381', 'RP1382', 'RP1383', 'RP1384', 'RP1385', 'RP1386', 'RP1388', 'RP1434', 'RP1436', 'RP1437', 'RP1438', 'RP1439', 'RP1441', 'RP1442', 'RP1460', 'RP1463', 'RP1464', 'RP1465', 'RP1466', 'RP1467', 'RP1468', 'RP1470', 'RP1491', 'RP1517', 'RP1520', 'RP1521', 'RP1531', 'RP1532', 'RP1533', 'RP1534', 'RP1549', 'RP1575', 'RP1576', 'RP1577', 'RP1578', 'RP1579']}\n",
      "\n",
      "Kmeans clustering: iter=2, K=3, mincov=0.55, minmap={0: 0.2, 1: 0.2, 2: 0.2}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/dss/dsshome1/0A/ra93rey/.conda/envs/ipyrad/lib/python3.10/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Samples: 184\n",
      "Sites before filtering: 7238146\n",
      "Filtered (indels): 143258\n",
      "Filtered (bi-allel): 222874\n",
      "Filtered (mincov): 7217752\n",
      "Filtered (minmap): 7133175\n",
      "Filtered (subsample invariant): 375268\n",
      "Filtered (minor allele frequency): 0\n",
      "Filtered (combined): 7222276\n",
      "Sites after filtering: 17025\n",
      "Sites containing missing values: 17025 (100.00%)\n",
      "Missing values in SNP matrix: 949313 (30.30%)\n",
      "SNPs (total): 17025\n",
      "SNPs (unlinked): 534\n",
      "Imputation: 'sampled'; (0, 1, 2) = 94.4%, 4.7%, 0.9%\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/dss/dsshome1/0A/ra93rey/.conda/envs/ipyrad/lib/python3.10/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{0: ['RP1210', 'RP1211', 'RP1212', 'RP1213', 'RP1221', 'RP1239', 'RP1304', 'RP1305', 'RP1312', 'RP1313', 'RP1314', 'RP1315', 'RP1316', 'RP1317', 'RP1338', 'RP1339', 'RP1340', 'RP1341', 'RP1342', 'RP1343', 'RP1379', 'RP1380', 'RP1381', 'RP1382', 'RP1383', 'RP1384', 'RP1385', 'RP1386', 'RP1388', 'RP1434', 'RP1436', 'RP1437', 'RP1438', 'RP1439', 'RP1441', 'RP1442', 'RP1460', 'RP1463', 'RP1464', 'RP1465', 'RP1466', 'RP1467', 'RP1468', 'RP1470', 'RP1491', 'RP1517', 'RP1520', 'RP1521', 'RP1531', 'RP1532', 'RP1533', 'RP1534', 'RP1549', 'RP1575', 'RP1576', 'RP1577', 'RP1578', 'RP1579'], 1: ['RP1230', 'RP1231', 'RP1232', 'RP1233', 'RP1234', 'RP1238', 'RP1240', 'RP1253', 'RP1254', 'RP1255', 'RP1256', 'RP1257', 'RP1258', 'RP1259', 'RP1260', 'RP1261', 'RP1263', 'RP1264', 'RP1265', 'RP1266', 'RP1267', 'RP1268', 'RP1269', 'RP1270', 'RP1271', 'RP1272', 'RP1274', 'RP1275', 'RP1276', 'RP1277', 'RP1278', 'RP1279', 'RP1280', 'RP1281', 'RP1282', 'RP1283', 'RP1284', 'RP1285', 'RP1286', 'RP1287', 'RP1288', 'RP1289', 'RP1290', 'RP1291', 'RP1292', 'RP1293', 'RP1294', 'RP1295', 'RP1297', 'RP1298', 'RP1301', 'RP1302', 'RP1403', 'RP1404', 'RP1405', 'RP1406', 'RP1407', 'RP1417', 'RP1418', 'RP1419', 'RP1420', 'RP1421', 'RP1422', 'RP1423', 'RP1426', 'RP1427', 'RP1456', 'RP1458', 'RP1459', 'RP1461', 'RP1462', 'RP1479', 'RP1480', 'RP1481', 'RP1482', 'RP1483', 'RP1484', 'RP1485', 'RP1487', 'RP1488', 'RP1489', 'RP1490', 'RP1494', 'RP1495', 'RP1496', 'RP1497', 'RP1499', 'RP1508', 'RP1509', 'RP1510', 'RP1511', 'RP1512', 'RP1513', 'RP1530', 'RP1550', 'RP1551', 'RP1561', 'RP1562', 'RP1563', 'RP1564', 'RP1565', 'RP1581', 'RP1582', 'RP1583', 'RP1584'], 2: ['I1', 'I2', 'I3', 'M1', 'RP1361', 'RP1362', 'RP1363', 'RP1364', 'RP1365', 'RP1366', 'RP1369', 'RP1552', 'RP1553', 'RP1554', 'RP1555', 'RP1556', 'RP1569', 'RP1570', 'RP1571', 'RP1572', 'RP1573']}\n",
      "\n",
      "Kmeans clustering: iter=3, K=3, mincov=0.3750000000000001, minmap={0: 0.2, 1: 0.2, 2: 0.2}\n",
      "Samples: 184\n",
      "Sites before filtering: 7238146\n",
      "Filtered (indels): 143258\n",
      "Filtered (bi-allel): 222874\n",
      "Filtered (mincov): 7178619\n",
      "Filtered (minmap): 7133175\n",
      "Filtered (subsample invariant): 375268\n",
      "Filtered (minor allele frequency): 0\n",
      "Filtered (combined): 7195267\n",
      "Sites after filtering: 44034\n",
      "Sites containing missing values: 44034 (100.00%)\n",
      "Missing values in SNP matrix: 3683092 (45.46%)\n",
      "SNPs (total): 44034\n",
      "SNPs (unlinked): 1377\n",
      "Imputation: 'sampled'; (0, 1, 2) = 93.1%, 5.8%, 1.1%\n",
      "{0: ['RP1230', 'RP1231', 'RP1232', 'RP1233', 'RP1234', 'RP1238', 'RP1239', 'RP1240', 'RP1253', 'RP1254', 'RP1255', 'RP1256', 'RP1257', 'RP1258', 'RP1259', 'RP1260', 'RP1261', 'RP1263', 'RP1264', 'RP1265', 'RP1266', 'RP1267', 'RP1268', 'RP1269', 'RP1270', 'RP1271', 'RP1272', 'RP1274', 'RP1275', 'RP1276', 'RP1277', 'RP1278', 'RP1279', 'RP1280', 'RP1281', 'RP1282', 'RP1283', 'RP1284', 'RP1285', 'RP1286', 'RP1287', 'RP1288', 'RP1289', 'RP1290', 'RP1291', 'RP1292', 'RP1293', 'RP1294', 'RP1295', 'RP1297', 'RP1298', 'RP1301', 'RP1302', 'RP1403', 'RP1404', 'RP1405', 'RP1406', 'RP1407', 'RP1417', 'RP1418', 'RP1419', 'RP1420', 'RP1421', 'RP1422', 'RP1423', 'RP1426', 'RP1427', 'RP1456', 'RP1458', 'RP1459', 'RP1461', 'RP1462', 'RP1479', 'RP1480', 'RP1481', 'RP1482', 'RP1483', 'RP1484', 'RP1485', 'RP1487', 'RP1488', 'RP1489', 'RP1490', 'RP1494', 'RP1495', 'RP1496', 'RP1497', 'RP1499', 'RP1508', 'RP1509', 'RP1510', 'RP1511', 'RP1512', 'RP1513', 'RP1530', 'RP1550', 'RP1551', 'RP1561', 'RP1562', 'RP1563', 'RP1564', 'RP1565', 'RP1581', 'RP1582', 'RP1583', 'RP1584'], 1: ['I1', 'I2', 'I3', 'M1', 'RP1361', 'RP1362', 'RP1363', 'RP1364', 'RP1365', 'RP1366', 'RP1369', 'RP1552', 'RP1553', 'RP1554', 'RP1555', 'RP1556', 'RP1569', 'RP1570', 'RP1571', 'RP1572', 'RP1573'], 2: ['RP1210', 'RP1211', 'RP1212', 'RP1213', 'RP1221', 'RP1304', 'RP1305', 'RP1312', 'RP1313', 'RP1314', 'RP1315', 'RP1316', 'RP1317', 'RP1338', 'RP1339', 'RP1340', 'RP1341', 'RP1342', 'RP1343', 'RP1379', 'RP1380', 'RP1381', 'RP1382', 'RP1383', 'RP1384', 'RP1385', 'RP1386', 'RP1388', 'RP1434', 'RP1436', 'RP1437', 'RP1438', 'RP1439', 'RP1441', 'RP1442', 'RP1460', 'RP1463', 'RP1464', 'RP1465', 'RP1466', 'RP1467', 'RP1468', 'RP1470', 'RP1491', 'RP1517', 'RP1520', 'RP1521', 'RP1531', 'RP1532', 'RP1533', 'RP1534', 'RP1549', 'RP1575', 'RP1576', 'RP1577', 'RP1578', 'RP1579']}\n",
      "\n",
      "Kmeans clustering: iter=4, K=3, mincov=0.2, minmap={0: 0.2, 1: 0.2, 2: 0.2}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/dss/dsshome1/0A/ra93rey/.conda/envs/ipyrad/lib/python3.10/site-packages/sklearn/cluster/_kmeans.py:870: FutureWarning: The default value of `n_init` will change from 10 to 'auto' in 1.4. Set the value of `n_init` explicitly to suppress the warning\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Samples: 184\n",
      "Sites before filtering: 7238146\n",
      "Filtered (indels): 143258\n",
      "Filtered (bi-allel): 222874\n",
      "Filtered (mincov): 6981275\n",
      "Filtered (minmap): 7137761\n",
      "Filtered (subsample invariant): 375268\n",
      "Filtered (minor allele frequency): 0\n",
      "Filtered (combined): 7150186\n",
      "Sites after filtering: 89115\n",
      "Sites containing missing values: 89115 (100.00%)\n",
      "Missing values in SNP matrix: 9489079 (57.87%)\n",
      "SNPs (total): 89115\n",
      "SNPs (unlinked): 2829\n",
      "Imputation: 'sampled'; (0, 1, 2) = 91.5%, 7.2%, 1.3%\n"
     ]
    }
   ],
   "source": [
    "#Enter data file and parameters for pca\n",
    "# init (load data and apply filtering) pca object with input data and (optional) parameter options\n",
    "pca = ipa.pca(\n",
    "    data=data,\n",
    "    imap=imap,\n",
    "    minmap=minmap,\n",
    "    mincov=0.2, #filters SNPs that are shared across less than some proportion of all samples \n",
    "    impute_method=3, #kmeans imputation method with 3 clusters\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Subsampling SNPs: 2829/89115\n"
     ]
    }
   ],
   "source": [
    "#run PCA analysis. Each replicate subsample a different random set of unlinked SNPs each time.\n",
    "pca.run(nreplicates=55,seed=12345)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# store the PC axes as a dataframe\n",
    "df = pd.DataFrame(pca.pcaxes[0], index=pca.names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write the PC axes to a CSV file\n",
    "df.to_csv(\"pca_184indv.csv\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ipyrad",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.11"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
